android/core/src/main/java/com/stadiamaps/ferrostar/core/extensions/RouteExtensions.kt

```diff
@@ -0,0 +1,23 @@
+package com.stadiamaps.ferrostar.core.extensions
+
+import uniffi.ferrostar.Route
+import uniffi.ferrostar.createRouteFromOsrm
+
+/**
+ * Create a [Route] from OSRM route and waypoint data.
+ *
+ * This behavior uses the same internal decoders as the OsrmResponseParser. This function will
+ * automatically map & combine via and break waypoints from the route and waypoint data objects.
+ *
+ * @param route The encoded JSON data for the OSRM route.
+ * @param waypoints The encoded JSON data for the OSRM waypoints.
+ * @param polylinePrecision The polyline precision.
+ * @return The navigation [Route]
+ */
+fun Route.Companion.fromOsrm(
+    route: ByteArray,
+    waypoints: ByteArray,
```

Comment:
Is a caller of this function (and the Swift version and the underlying Rust) likely to have waypoints as raw bytes? Or as something higher level? I honestly don't know since I'm not a likely user of it.

Reply:
Discussed on the call.